lib/projections/ecto.ex

```diff
@@ -107,6 +107,65 @@ defmodule Commanded.Projections.Ecto do
         end
       end
 
+      def update_projection_batch(events, multi_fn) do
+        {first_event, first_event_metadata} = List.first(events)
+
+        first_event_number = Map.fetch!(first_event_metadata, :event_number)
+        projection_name = Map.fetch!(first_event_metadata, :handler_name)
```

Comment:
Minor:

Since we're crashing if the list is non-empty or if the event metadata doesn't have the expected keys, we could do all of this in a fun head match. Or a mix of the two.

What about:

```
def update_projection_batch([{first_event, first_event_metadata} |_] = events, multi_fn) do
    %{event_number: first_event_number, handler_name: projection_name} = first_event_metadata
...
```